packages/block-library/src/navigation/index.php

```diff
@@ -619,14 +619,34 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	);
 
 	$should_display_icon_label = isset( $attributes['hasIcon'] ) && true === $attributes['hasIcon'];
-	$toggle_button_icon        = '<svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" focusable="false"><rect x="4" y="7.5" width="16" height="1.5" /><rect x="4" y="15" width="16" height="1.5" /></svg>';
+	$toggle_button_icon        = apply_filters(
```

Comment:
Thanks for picking this up.

The problem with filtering here is that the user could have selected a particular icon in the editor, but then a filter applied by a plugin renders something completely different on the frontend, which results in a very confusing experience.

I think ideally the solution would be a filter that modifies the array of available icons for the user to select.

That may require JS and PHP code though.

Reply:
You're right, or replace filters with making possible to upload custom icons on Gutenberg.